werkzeug/contrib/cache.py

```diff
@@ -658,7 +674,7 @@ def _prune(self):
                     remove = False
                     with open(fname, 'rb') as f:
                         expires = pickle.load(f)
-                    remove = expires <= now or idx % 3 == 0
+                    remove = (expires != 0 and expires <= now) or idx % 3 == 0
```

Comment:
I agree, I also thought it was unnecessary, until the unit test started failing. I guess that's why you should also add tests :smile: